File: server/tests/fixtures/alpha_rule_triggers/csharp/test_flaky_sleep.cs
// Should trigger: test.flaky_sleep
using NUnit.Framework;
using System.Threading;

[TestFixture]
public class FlakyTest {
    [Test]
    public void TestWithSleep() {
        Thread.Sleep(1000);  // Flaky!
        Assert.Pass();
    }
}

File: server/tests/fixtures/alpha_rule_triggers/javascript/test_flaky_sleep.js
// Should trigger: test.flaky_sleep
describe('Flaky Test', () => {
    it('should wait', async () => {
        startProcess();
        await new Promise(r => setTimeout(r, 5000));  // flaky sleep
        expect(checkResult()).toBe(true);
    });
});

function startProcess() {}
function checkResult() { return true; }
